ENIGMAsystem/SHELL/Bridges/General/DX9Context.h

```diff
@@ -106,404 +106,402 @@ void LoadState() {
+  device->SetSoftwareVertexProcessing(bSoftware);
 }
 
 void SetMaterial(const D3DMATERIAL9 *pMaterial) {
-	device->SetMaterial(pMaterial);
+  device->SetMaterial(pMaterial);
 }
 
 void SetFVF(DWORD FVF) {
-	//EndShapesBatching();
-	device->SetFVF(FVF);
+  //EndShapesBatching();
+  device->SetFVF(FVF);
 }
 
 void SetViewport(const D3DVIEWPORT9 *pViewport) {
-	EndShapesBatching();
-	device->SetViewport(pViewport);
+  EndShapesBatching();
+  device->SetViewport(pViewport);
 }
 
 void SetLight(DWORD Index, const D3DLIGHT9 *pLight) {
-device->SetLight( Index, pLight );
-	return;
-	// Update the light state cache
-    // If the return value is 'true', the command must be forwarded to the D3D Runtime.
-	map< DWORD, D3DLIGHT9 >::iterator it = cacheLightStates.find( Index );
-    if ( cacheLightStates.end() == it ) {
-        cacheLightStates.insert( map< DWORD, D3DLIGHT9 >::value_type(Index, *pLight) );
-		EndShapesBatching();
-        device->SetLight( Index, pLight );
-		return;
-    }
-    //if( it->second == *pLight )
-        //return;
-    it->second = *pLight;
-	EndShapesBatching();
+  device->SetLight( Index, pLight );
+  return;
+  // Update the light state cache
+  // If the return value is 'true', the command must be forwarded to the D3D Runtime.
+  map< DWORD, D3DLIGHT9 >::iterator it = cacheLightStates.find( Index );
+  if ( cacheLightStates.end() == it ) {
+    cacheLightStates.insert( map< DWORD, D3DLIGHT9 >::value_type(Index, *pLight) );
+    EndShapesBatching();
     device->SetLight( Index, pLight );
+    return;
+  }
+  //if( it->second == *pLight )
+    //return;
+  it->second = *pLight;
+  EndShapesBatching();
+  device->SetLight( Index, pLight );
 }
 
 void SetTransform(D3DTRANSFORMSTATETYPE State, const D3DMATRIX *pMatrix) {
-	// if projection EndShapesBatching();
-	EndShapesBatching();
-	device->SetTransform(State, pMatrix);
+  // if projection EndShapesBatching();
+  EndShapesBatching();
+  device->SetTransform(State, pMatrix);
 }
 
 void SetRenderState(D3DRENDERSTATETYPE State, DWORD Value) {
-	// Update the render state cache
-    // If the return value is 'true', the command must be forwarded to the D3D Runtime.
-	map< D3DRENDERSTATETYPE, DWORD >::iterator it = cacheRenderStates.find( State );
-    if ( cacheRenderStates.end() == it ) {
-        cacheRenderStates.insert( map< D3DRENDERSTATETYPE, DWORD >::value_type(State, Value) );
-		EndShapesBatching();
-        device->SetRenderState( State, Value );
-    }
-    if( it->second == Value )
-        return;
-    it->second = Value;
-	EndShapesBatching();
+  // Update the render state cache
+  // If the return value is 'true', the command must be forwarded to the D3D Runtime.
+  map< D3DRENDERSTATETYPE, DWORD >::iterator it = cacheRenderStates.find( State );
+  if ( cacheRenderStates.end() == it ) {
+    cacheRenderStates.insert( map< D3DRENDERSTATETYPE, DWORD >::value_type(State, Value) );
+    EndShapesBatching();
     device->SetRenderState( State, Value );
```

Comment:
I put it in Mr. Lightbeer, it's supposed to manage the state because I couldn't get D3D9's state manager working or something retarded I was thinking at the time. But yeah the whole thing probably shouldn't exist. Don't write ENIGMA code and do drugs, it doesn't mix well.